File: source_funcs/go_transcoder/FIND_THREE_ELEMENT_FROM_DIFFERENT_THREE_ARRAYS_SUCH_THAT_THAT_A_B_C_K/FIND_THREE_ELEMENT_FROM_DIFFERENT_THREE_ARRAYS_SUCH_THAT_THAT_A_B_C_K.go
package main

func f_gold(a1 []int, a2 []int, a3 []int, n1 int, n2 int, n3 int, sum int) bool {
	for i := int(0); i < n1; i++ {
		for j := int(0); j < n2; j++ {
			for k := int(0); k < n3; k++ {
				if a1[i]+a2[j]+a3[k] == sum {
					return true
				}
			}
		}
	}
	return false
}

File: rust_funcs/cpp_transcoder/FIND_THREE_ELEMENT_FROM_DIFFERENT_THREE_ARRAYS_SUCH_THAT_THAT_A_B_C_K/FIND_THREE_ELEMENT_FROM_DIFFERENT_THREE_ARRAYS_SUCH_THAT_THAT_A_B_C_K.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32;

fn f_gold(a1: [f32;2], a2: [f32;2], a3: [f32;2], n1: f32, n2: f32, n3: f32, sum: f32) -> i32 {
    for i in 0..n1 as usize {
        for j in 0..n2 as usize {
            for k in 0..n3 as usize {
                if a1[i] + a2[j] + a3[k] == sum {
                    return 1.0;
                }
            }
        }
    }
    0.0
}
